lib/applications/field-resolvers.ts

```diff
@@ -43,11 +44,55 @@ export const applicationApplicantResolver: FieldResolver<
 
 /**
  * Fetch processing data of application
+ * Refresh temporary s3 application document url if it has expired.
  * @returns Application processing object
  */
 export const applicationProcessingResolver: FieldResolver<
   Application,
   Omit<ApplicationProcessing, 'invoice'>
 > = async (parent, _args, { prisma }) => {
-  return await prisma.application.findUnique({ where: { id: parent.id } }).applicationProcessing();
+  const application = await prisma.application
+    .findUnique({ where: { id: parent.id } })
+    .applicationProcessing();
+
+  if (!application) {
+    return null;
+  }
+
+  if (!process.env.APPLICATION_DOCUMENT_LINK_TTL_DAYS) {
+    throw new ApolloError('Application document link duration not defined');
+  }
+  // Update the signed S3 URL if it has expired.
+  // Get the valid duration period from env.
+  const applicationDocumentLinkDuration = parseInt(process.env.APPLICATION_DOCUMENT_LINK_TTL_DAYS);
+  const DAY = 24 * 60 * 60 * 1000;
+  const daysDifference =
+    Math.floor(new Date().getTime() / DAY) -
+    Math.floor(application?.documentsUrlUpdatedAt.getTime() / DAY);
+  if (daysDifference > applicationDocumentLinkDuration) {
+    let signedUrl;
+    try {
+      signedUrl = getSignedUrlForS3(application.documentsS3ObjectKey as string);
+    } catch (error) {
+      throw new ApolloError(`Failed to get application document URL: ${error}`);
+    }
+    try {
+      await prisma.application.update({
+        where: { id: application.id },
+        data: {
+          applicationProcessing: {
+            update: {
+              documentsUrl: signedUrl,
+            },
+          },
+        },
+      });
+    } catch {
+      throw new ApolloError('Failed to update temporary application document URL');
+    }
+    return await prisma.application
+      .findUnique({ where: { id: parent.id } })
+      .applicationProcessing();
+  }
+  return application;
```

Comment:
i think you should be able to do something like this to keep typescript happy

```suggestion
  return {
    ...application,
    documentsUrl: signedUrl
  };
```

Reply:
I get a slight type error where string | undefined cannot be of type Maybe<String>. To fix this I let `documentsUrl: signedUrl || null`. Not sure if this is correct I am assuming documentsUrl can be null for when there is no document uploaded.